### ISSUE TITLE
QCSubmit CI is segfaulting

### ISSUE THREAD

**Issue by j-wags** [openforcefield/openff-qcsubmit#278]
Segfaulting test seems to be `openff/qcsubmit/_tests/test_datasets.py::test_dataset_to_pdf_no_torsions[Openeye by default.]`

https://github.com/openforcefield/openff-qcsubmit/actions/runs/8697856360/job/23853864399#step:7:193

OE hasn't pushed a new package recently ([last main label package was 3 months ago](https://anaconda.org/OpenEye/OpenEye-toolkits/files?channel=main))

**Comment by mattwthompson** [comment]
Having dealt with similar (-ly cryptic) failures in the past, I'd try to parry down which modules that are bundled with OpenEye and might have had releases recently. (I quickly checked a few of the typical suspects and nothing was obvious.)

**Comment by mattwthompson** [comment]
If it helps narrow it down, it's likely in the OEDepict import, so maybe something changed with a visualization-adjacent upstream. (It'd be much messier if this is not related to a visualization tool.)

**Comment by j-wags** [comment]
AFAICT the openeye conda packages vendor everything. They only have python and pytest as deps. Am I overlooking a source of outside code?

<img width="1122" alt="Screenshot 2024-04-16 at 11 57 38 AM" src="https://github.com/openforcefield/openff-qcsubmit/assets/16329175/e9556bdd-1288-4f3a-850f-bb00e1323f68">

**Comment by mattwthompson** [comment]
Nothing sticks out with naive inspection of the changes in imported Python modules:

```python
In [1]: import sys

In [2]: mod1 = set(sys.modules.keys())

In [3]: from openeye import oechem

In [4]: mod2 = set(sys.modules.keys())

In [5]: from openeye import oedepict

In [6]: mod3 = set(sys.modules.keys())

In [7]: mod1.difference(mod2)
Out[7]: set()

In [8]: mod2.difference(mod1)
Out[8]:
{'_oechem4003',
 'importlib.abc',
 'importlib.resources',
 'importlib.resources._adapters',
 'importlib.resources._common',
 'importlib.resources._legacy',
 'importlib.resources.abc',
 'libs',
 'libs.openeye_platform',
 'oechem',
 'openeye',
 'openeye.oechem',
 'swig_runtime_data4'}

In [9]: mod3.difference(mod1)
Out[9]:
{'_oechem4003',
 '_oedepict2533',
 'importlib.abc',
 'importlib.resources',
 'importlib.resources._adapters',
 'importlib.resources._common',
 'importlib.resources._legacy',
 'importlib.resources.abc',
 'libs',
 'libs.openeye_platform',
 'oechem',
 'oedepict',
 'openeye',
 'openeye.oechem',
 'openeye.oedepict',
 'swig_runtime_data4'}

In [10]: mod3.difference(mod2)
Out[10]: {'_oedepict2533', 'oedepict', 'openeye.oedepict'}

**Comment by j-wags** [comment]
I can reproduce the error locally and downgrading to openff-toolkit 0.15.2 fixes it 😿 Time to do some digging.

**Comment by mattwthompson** [comment]
I bisected it to https://github.com/openforcefield/openff-toolkit/pull/1798

This commit is fine: https://github.com/openforcefield/openff-toolkit/commit/b739698d4187943ac8e31c59a0866b60247c6ed5

I can look into this more closely tomorrow if you don't get to the bottom of it today - sorry for introducing a regression.

**Comment by j-wags** [comment]
Hah, no worries. Great sleuthing. It's a weird enough error that normal CI didn't catch it. Looking into it, I'll let you know what I find.

**Comment by mattwthompson** [comment]
My suggestion earlier wasn't the right path to go down, but in case we go down it once again, here's a better starting point for getting what might be a part of compiled OpenEye code: https://docs.eyesopen.com/applications/license_3rdparty.html

**Comment by j-wags** [comment]
I think the actual problem might be that qcsubmit [directly sets](https://github.com/openforcefield/openff-qcsubmit/blob/35fe7724e010a908aa4f2640676249f37b5b6d63/openff/qcsubmit/utils/visualize.py#L78) `offmol._conformers = []`, but OFFTK might expect "no conformers" to only be represented by `_conformers = None`. Testing that out now...

**Comment by j-wags** [comment]
Resolved by #277
